tests/luagit2_commit_test.lua

```diff
@@ -0,0 +1,234 @@
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	author_signature = lib.luagit2_commit_author(commit)
+	lunatest.assert_not_nil(author_signature)
+	lunatest.assert_equal("Satyendra_Kumar_Banjare" , lib.luagit2_get_signature_details(author_signature,1))
+	lunatest.assert_equal("satyendrabanjare99@gmail.com" , lib.luagit2_get_signature_details(author_signature,2))
+end
+
+function luagit2_commit_tests.test_luagit2_commit_body()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id_nil_body = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id_nil_body)
+
+	commit_with_nil_body = lib.luagit2_commit_lookup(repo,commit_id_nil_body)
+	lunatest.assert_not_nil(commit_with_nil_body)
+
+	commit_body_is_nil = lib.luagit2_commit_body(commit_with_nil_body)
+	lunatest.assert_nil(commit_body_is_nil)
+
+	commit_id_non_nil_body = lib.luagit2_oid_fromstr("15a746e5e897349f7fd9a3e7308eb0c7c7e0c010")
+	lunatest.assert_not_nil(commit_id_non_nil_body)
+
+	commit_with_non_nil_body = lib.luagit2_commit_lookup(repo,commit_id_non_nil_body)
+	lunatest.assert_not_nil(commit_with_non_nil_body)
+
+	commit_body_is_not_nil = lib.luagit2_commit_body(commit_with_non_nil_body)
+	lunatest.assert_not_nil(commit_body_is_not_nil)
+	lunatest.assert_string(commit_body_is_not_nil)
+
+end
+
+function luagit2_commit_tests.test_luagit2_commit_committer()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	committer_signature = lib.luagit2_commit_committer(commit)
+	lunatest.assert_not_nil(committer_signature)
+	lunatest.assert_equal("Satyendra_Kumar_Banjare" , lib.luagit2_get_signature_details(committer_signature,1))
+	lunatest.assert_equal("satyendrabanjare99@gmail.com" , lib.luagit2_get_signature_details(committer_signature,2))
+end
+
+function luagit2_commit_tests.test_luagit2_commit_message()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	commit_message = lib.luagit2_commit_message(commit)
+	lunatest.assert_not_nil(commit_message)
+	lunatest.assert_string(commit_message)
```

Comment:
How do you know it's the message you expect?

Reply:
One thing you can do is use luagit2 itself to create a commit from variables before the tests. That way you can compare it with the message stored.

Reply:
Ok!